Display Effect dropdown for turned off lights

OleksandrBerchenko opened the pull request:
## Description:

**Related issue:** fixes https://github.com/home-assistant/home-assistant/issues/20583

Display Effect dropdown for all lights regardless of state (not only for turned on lights).

Related PR into `home-assistant` repo: https://github.com/home-assistant/home-assistant/pull/20750

OleksandrBerchenko committed:
Display Effect dropdown for turned off lights

balloob commented:
We shouldn't merge this. Effects are an edge case and should not clutter the more info dialog when a light is off.

OleksandrBerchenko commented:
Well, I agree that effects are an edge case and most lights probably don't provide any effects.  In that case Effect dropdown is never displayed.  But for the rest, effect is a part of the "turn on" routine and I assume that in most cases you want to activate an effect when the light is off.  That makes the dropdown even more helpful for turned off lights.  At the moment to activate an effect you need 1. turn on the light or a group of lights with its default configuration (not what you need right now), 2. apply the actual effect.

I'm currently working on adding support for Osram Lightify scenes to Home Assistant.  They are effects applied to a group of lights.  I have plenty of scenes at my home and in 90% of situations I use them when a group of lights is completely off.

Does it makes sense?

Thanks!

amelchio commented:
That is how the UI works. For example, you also have to turn on the light before you can set a different color. You can of course argue that this is a bad interface but singling out effects does not seem warranted.

OleksandrBerchenko commented:
@amelchio Thanks for your comment.  I believe that depends on exact use cases of the users.

Personally I never needed to change a color or temperature from UI (I configured each lamp once and that's all).  But if I ever need, I can rather imagine a situation like "the light is already turned on, but I'm bored with its color/temperature, so I want to change it" and not like "the light is turned off, and I want to turn it on with some particular color/temperature in mind".  In other words, for me it's totally ok to display color/temperature controls only when a light is turned on.

On the other hand, when controlling my Osram Lightify lights, I mostly rely on scenes (effects in terms of HA).  I activate them many times a day, and in 80% cases that happens when a light is turned off.  So, for my experience current behavior is not very handy and it would be great to have an ability to change it.

Thanks!

amelchio commented:
Scenes? The Osram Lightify platform seems to list only the [random effect](https://github.com/home-assistant/home-assistant/blob/0.86.3/homeassistant/components/light/osramlightify.py#L161).

Are we talking about the same thing?

OleksandrBerchenko commented:
Yes :)  I have updated the Osram Lightify platform with many fixes and improvements, including support for scenes (finally!)  At the moment I'm finishing local testing and plan to create a PR soon.

If you are interested, I have also reworked the underlying library - see https://github.com/tfriedel/python-lightify.  The new version is already on PyPi.

Thanks!
